### PR TITLE
feat: add statitics service

### PR DESCRIPTION
**Pull request by ShrBox**
## What does this PR do?

Add statitics service

## Which issues does this PR resolve?

Nope

## Checklist before merging

Thank you for your contribution to the repository. 
Before submitting this PR, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] Your code follows the code style of this repository (see the wiki)
- [x] You have tested all functions
- [x] You have not used code without license
- [x] You have added statement for third-party code

### CODE REVIEW

**Comment by OEOTYAN** [comment]
真是太屎了 就为这b东西还得加俩库 尤其是uuid最屎 明明有uuid类

File: xmake.lua [hunk 1]
```diff
@@ -106,6 +109,7 @@ target("LeviLamina")
         "pcg_cpp",
         "pfr",
         "symbolprovider",
+        "cpp-httplib",
```
Comment (RimuruChan) [inline xmake.lua]: 内部依赖，不需要的暴露

File: src/ll/core/tweak/Statistics.cpp [hunk 2]
```diff
@@ -0,0 +1,237 @@
+#include "ll/core/tweak/Statistics.h"
```
Comment (RimuruChan) [inline src/ll/core/tweak/Statistics.cpp:1]: 导入应当分组

File: src/ll/core/tweak/Statistics.cpp [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            return true;
+        });
+    }
+    pluginsJson.emplace_back(addAdvancedPie("player_platform", platforms));
+
+    return pluginsJson;
+}
+
+// generate random float between 0.0 and 1.0
+float randomFloat() {
```
Comment (RimuruChan) [inline src/ll/core/tweak/Statistics.cpp]: 应当使用RandomUtils

File: src/ll/core/tweak/Statistics.cpp [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    std::random_device                    rd;
+    std::mt19937                          gen(rd());
+    std::uniform_real_distribution<float> dis(0.0, 1.0);
+    return dis(gen);
+}
+
+std::string generateUuidV4() {
+    static std::random_device              rd;
```
Comment (RimuruChan) [inline src/ll/core/tweak/Statistics.cpp]: >  _mc不带随机uuid工具吗？_

File: src/ll/core/tweak/Statistics.cpp [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+    // ll::logger.debug("Submit: {}", json.dump());
+    try {
+        auto result = client.Post("/api/v2/data/bukkit", headers, json.dump(), "application/json");
+        // ll::logger.debug("Body: {} Status: {}", result->body, result->status);
+    } catch (const std::exception& e) {
+        // ll::error_utils::printCurrentException(ll::logger);
+    }
+}
+
+void InitStatitics() {
```
Comment (RimuruChan) [inline src/ll/core/tweak/Statistics.cpp]: 不符合命名标准

File: src/ll/core/tweak/Statistics.cpp [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+    if (!std::filesystem::exists(plugin::getPluginsRoot() / u8"LeviLamina/data")) {
```
Comment (RimuruChan) [inline src/ll/core/tweak/Statistics.cpp]: 过长namespace，善用using，以及namespace别名，如 `namespace fs = std::filesystem;`

File: src/ll/core/tweak/Statistics.h [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <memory>
+
+namespace ll {
+
+struct Statitics {
+    struct Impl;
+    std::unique_ptr<Impl> impl;
```
Comment (RimuruChan) [inline src/ll/core/tweak/Statistics.h]: 不必要的pimpl封装

File: xmake.lua [hunk 8]
```diff
@@ -15,6 +15,8 @@ add_requires("magic_enum")
 add_requires("nlohmann_json")
 add_requires("rapidjson v1.1.0")
 add_requires("mimalloc")
+add_requires("openssl 1.1.1-w")
+add_requires("cpp-httplib", {configs={ssl=true, zlib=true}})
```
Comment (RimuruChan) [inline xmake.lua:20]: zlib可能需要重写禁用系统库，否则有msys2的环境会出问题

**Review by RimuruChan — CHANGES_REQUESTED** [review]

File: src/ll/core/tweak/Statistics.h (hunk 7 above)
Comment (ShrBox) [reply src/ll/core/tweak/Statistics.h]: 源文件里面有个hook用到了

Comment (RimuruChan) [reply src/ll/core/tweak/Statistics.h:9]: 构造函数static 存一个就行了

**Review by RimuruChan — CHANGES_REQUESTED** [review]

File: src/ll/core/tweak/Statistics.h (hunk 7 above)
Comment (RimuruChan) [reply src/ll/core/tweak/Statistics.h]: 没懂，这需要pimpl吗，放构造函数static不行吗

Comment (ShrBox) [reply src/ll/core/tweak/Statistics.h]: > 没懂，这需要pimpl吗，放构造函数static不行吗

好

**Review by RimuruChan — APPROVED** [review]

**Pull request merged**